doc/releases/release-notes-3.5.rst

```diff
@@ -100,6 +100,11 @@ Architectures
 
   * Added basic MMU v2 Support.
 
+* X86
```

Comment:
double nit: "x86" and it would have been nice to have all archs sorted alphabetically? (note that it was already not the case before this change).
Maybe something to keep in mind to fix during the final cleanup, if there is one.

Reply:
Yeah, I was going back and forth between x86 and X86. We have "Intel Platforms (X86)" in MAINTAINERS.yml which should probably be fixed to lower case.

Reply:
Now updated (the x86 case issue, let's look at the ordering separately)